### PR TITLE
SREP-1539: Add Machine API Exporter Prometheus Rule

### PR DESCRIPTION
**Pull request by lisa**
Note: KCS document link to be provided once it is written.

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

### CODE REVIEW

File: deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    prometheus: sre-machine-api-node-mismatch
+    role: alert-rules
+  name: sre-machine-api-node-mismatch
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: sre-machine-api-node-mismatch
+    rules:
+    - alert: NoNodeObjectForMachineWarning
+      expr: &promql machine_api_status == 0
```
Comment (lisa) [inline deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml]: Note: `&promql` and `kcslink` (later on line 20) establish two YAML anchors that are used on lines 21 and 26 respectively to write only one time the PromQL for this rule and the KCS link for the remediation document. The goal is to DRY out this file

File: deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  labels:
+    prometheus: sre-machine-api-node-mismatch
+    role: alert-rules
+  name: sre-machine-api-node-mismatch
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: sre-machine-api-node-mismatch
+    rules:
+    - alert: NoNodeObjectForMachineWarning
```
Comment (jewzaam) [inline deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml]:
```suggestion
    - alert: NoNodeObjectForMachineWarningSRE
```

File: deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      expr: &promql machine_api_status == 0
+      for: 30m
+      severity: warning
+      annotations:
+        message: "The Machine object {{ $labels.machine_name }} hasn't had an associated Node for the past 30 minutes."
+        link: &kcslink "https://access.redhat.com"
+    - alert: NoNodeObjectForMachineCritical
```
Comment (jewzaam) [inline deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml]:
```suggestion
    - alert: NoNodeObjectForMachineCriticalSRE
```

File: deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    role: alert-rules
+  name: sre-machine-api-node-mismatch
+  namespace: openshift-monitoring
+spec:
+  groups:
+  - name: sre-machine-api-node-mismatch
+    rules:
+    - alert: NoNodeObjectForMachineWarning
+      expr: &promql machine_api_status == 0
+      for: 30m
```
Comment (jewzaam) [inline deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml]:
```suggestion
      for: 15m
```
I assume provisioning a node might take 10 minutes based on what I've observed.  15 min seems a good time to warn.

File: deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      severity: warning
+      annotations:
+        message: "The Machine object {{ $labels.machine_name }} hasn't had an associated Node for the past 30 minutes."
+        link: &kcslink "https://access.redhat.com"
+    - alert: NoNodeObjectForMachineCritical
+      expr: *promql
+      for: 60m
```
Comment (jewzaam) [inline deploy/sre-prometheus/100-machine-api.PrometheusRule.yaml]:
```suggestion
      for: 30m
```
I'd like to bring this in, but not sure how far we can.  I think doubling the warning time is a reasonable start.

**Comment by RiRa12621** [comment]
I think this should be reverted. The alert is firing for no reason so we have a high risk of creating alert fatigue.

**Comment by jewzaam** [comment]
@RiRa12621 created https://jira.coreos.com/browse/SREP-1852 for this issue.

**Pull request merged**